### PR TITLE
Update/complex commanding

### LINKED ISSUE #38: Add in Commanding With Complex Types
| | |
|:---|:---|
|**_F´ Version_**| |
|**_Affected Component_**|  |
---
## Feature Description

https://github.com/nasa/fprime/issues/607

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|  |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Builds Without Errors (y/n)_**|  |
|**_Unit Tests Pass (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

This allows commanding with complex argument types (struct, arrays, and arbitrary combinations of the two). By extension this means that parameters with complex types are also supported.

Additionally, some minor clean-up was done:
- Added help to settings page
- Refactored commanding into an "add-on" per modern F´ style
- Refactored commanding implementations
- Separation of concerns

## Rationale

F´ nominally supports these thing, however; the GDS arbitrarily limits their usage. Therefore, the GDS should be updated to support this.

## Testing/Review Recommendations

Run with associated PR in F´ that adds in a typeDemo.  Will comment back with PR number.

## Future Work

None.

### CODE REVIEW

**Comment by LeStarch** [comment]
@thomas-bc and @timcanham this is the big PR that formally allows complex types in our commanding and parameters.  Let me know how I might aid in your reviews and development.

**Comment by LeStarch** [comment]
For testing, use nasa/fprime#1937

**Comment by LeStarch** [comment]
Note: this does not fix/update seq-gen.  That work is outstanding.

**Comment by LeStarch** [comment]
Also fixes: nasa/fprime#602

**Review by timcanham — DISMISSED** [review]
Reviewed GUI live behavior. Not in depth code review.

File: src/fprime_gds/flask/static/js/validate.js [hunk 1]
```diff
@@ -47,50 +48,133 @@ export function find_case_insensitive(token, possible) {
...
+        return false;
+    } else {
+        argument.value = valid_arg;
+        argument.error = "";
+        return true;
+    }
     argument.error = "";
-    // Integral types checking
-    if (type.name in TYPE_LIMITS) {
+    return true;
```
Comment (thomas-bc) [inline src/fprime_gds/flask/static/js/validate.js:66]: These last 2 lines are unreachable. Is that intended?

**Review by thomas-bc — DISMISSED** [review]
Code review. Looks good to me.

**Review by thomas-bc — APPROVED** [review]

**Pull request merged**